### ISSUE TITLE
If a user does not exist but the app sharing directory does the UsrRouter will crash

### ISSUE THREAD

**Issue by ericfranz** [OSC/ood-dashboard#415]
```
#<ArgumentError: can't find user for awe0025>

/var/www/ood/apps/sys/dashboard/app/apps/usr_router.rb:23:in `getpwnam'
/var/www/ood/apps/sys/dashboard/app/apps/usr_router.rb:23:in `owner_title'
/var/www/ood/apps/sys/dashboard/app/apps/usr_router.rb:19:in `category'
/var/www/ood/apps/sys/dashboard/app/apps/ood_app.rb:154:in `category'
/var/www/ood/apps/sys/dashboard/app/apps/ood_app_group.rb:22:in `block in groups_for'
/var/www/ood/apps/sys/dashboard/app/apps/ood_app_group.rb:21:in `each'
/var/www/ood/apps/sys/dashboard/app/apps/ood_app_group.rb:21:in `group_by'
/var/www/ood/apps/sys/dashboard/app/apps/ood_app_group.rb:21:in `groups_for'
/var/www/ood/apps/sys/dashboard/app/controllers/apps_controller.rb:18:in `featured'
```

**Comment by ericfranz** [comment]
```
irb(main):002:0> require 'etc'
=> true
irb(main):004:0> Etc.getpwnam 'efranz'
=> #<struct Etc::Passwd name="efranz", passwd="*", uid=10851, gid=5515, gecos="Eric Franz", dir="/users/PZS0562/efranz", shell="/bin/bash">
irb(main):005:0> Etc.getpwnam 'efranzz'
ArgumentError: can't find user for efranzz
	from (irb):5:in `getpwnam'
	from (irb):5
	from /opt/rh/rh-ruby24/root/usr/bin/irb:11:in `<main>'
```

Write a test with an unlikely username.

**Comment by ericfranz** [comment]
Fixed 10fa0f477feea3235158bd5deca78e71f1c78298
